### PR TITLE
Re-implement sky fog from vanilla

### PR DESCRIPTION
**Pull request by b3spectacled**
Hiya, this mod appears to overwrite the vanilla behavior for applying fog to the sky, so I re-added it.

Vanilla applies a fog layer for both the sky and terrain.  Custom Fog currently does not render the former at all, so the horizon appears as a distinct band, much more noticeably with very far fog starts (or disabled fog).  Here's an image album with comparisons: https://imgur.com/a/wnRik9G

Thanks for considering!

### CODE REVIEW

**Comment by Setadokalo** [comment]
very nice! I'd like to check the code myself (tomorrow, unfortunately - very tired tonight) to see if there's a better way to handle it than setting the fog distance uniformly like this for the sky; I'm mostly just curious if there's a way to remove the banding effect without hardcoding the sky fog.

**Comment by Setadokalo** [comment]
well regardless of if there's some Magic Better Way™, this is definitely an improvement over the way it works now; my only complaint is that your implementation currently sets the fogStart, fogEnd, and fogMode to the vanilla sky settings, which is what they're already set to just before the mixin is invoked. In fact, we could return early all the way back during the first line in the `setFogFalloff()` method, which would both make it behave itself more if some other mod also mixes into `BackgroundRenderer#applyFog` and changes the skybox fog, and be just a bit faster (since it can avoid getting the config when rendering a skybox).
TLDR: I think it would be worthwhile to move the `if (fogType == SKY)` check to the first line of `setFogFalloff`.

**Comment by b3spectacled** [comment]
Thanks for the feedback, done.  I had half a thought with the fogType check in `changeFalloff`, was thinking one might be able to allow users to also set the fog mode for the sky fog; but it would probably be a fair amount of trouble for a relatively small visual effect that would probably be obscured by the terrain fog on any non-zero fog start or density.

**Comment by Setadokalo** [comment]
looks good! I'm gonna merge it today, and I'll make a new release within the next couple days - there was some other change I wanted to make first (that I've now forgotten :/ )

**Comment by b3spectacled** [comment]
Cool, thank you!  Appreciate the work you've done with this mod! :D
